reporter-influxdb/src/main/java/io/ultrabrew/metrics/reporters/influxdb/InfluxDBClient.java

```diff
@@ -57,8 +62,25 @@ private CloseableHttpClient getHttpClient() {
   private void doWrite(final String measurement, final String[] tags, final String[] fields,
       final long timestamp)
       throws IOException {
+    if (Strings.isNullOrEmpty(measurement)) {
+      LOGGER.warn("Null or empty measurement.");
+      return;
+    }
+    int rollback = byteBuffer.position();
     byteBuffer.put(measurement.getBytes(UTF_8));
     for (int i = 0; i < tags.length; i += 2) {
+      if (Strings.isNullOrEmpty(tags[i])) {
+        LOGGER.warn("Null or empty tag key in tags array: " 
+            + Arrays.toString(tags) + " for measurement " + measurement);
+        byteBuffer.position(rollback);
```

Comment:
Instead of adding the rollback to each check, it could be done with `try`-`finally` to make the code maybe a little less error prone, though I didn't actually try writing it to see if it would look better that way.

Reply:
I think that'd be a bit uglier as I don't want to rollback if we have valid data. Or I'd have to get the position and reset it to the same position in the finally.

Reply:
Or use `-1` or similar impossible position as marker for no rollback, yes. I'm just slightly concerned that this way it's entirely possible to forget the rollback call later when adding more checks. In the end it's a tradeoff, choosing which way produces the more readable and robust result.